lessons/git/intro/practice-code-2017-10-05.sh

```diff
@@ -0,0 +1,168 @@
+cat members.yml
+
+## Add yourself to the members list!
+## typing "nano filename" opens the file in the editor nano. 
+## you can use any editor you like, such as atom, notepad, vim, etc.
+nano members.yml
+
+## git status = check the status of your repository
+## you'll see a changed file
+git status
+
+## add = track and put the file into the git staging area (staging
+## just means ready to be saved into the history, but not actually
+## saved)
+git add members.yml
+git status ## just to see what happened
+
+## commit = save everything in the staging area into the history (the
+## git repo).  -m = the commit message to be included in the history.
+## Try to be descriptive with this message.
+git commit -m "Added myself to the members list"
+
+## log = check the saved history
+git log
+git status ## should be all clean again
+
+## Make some more edits to the members file
+nano members.yml
+git status
+
+## diff = compare the contents of the current file with the
+## (essentially) previous commit.
+git diff members.yml
+
+## When you add a filename to the commit command, git will save that
+## file directly into the history. This bypasses the staging area.
+git commit members.yml -m "Added a line to my bio"
+
+## Check your history of the saved commits.
+git log
+
+## Viewing history: git checkout
+
+## checkout = go back, forward, or around to different commits (or
+## branches) in your commit history.  c1361 is the (completely!!)
+## unique commit hashtag (you can see the hash when you do git log; no
+## one will have the same hash!). 
+
+## open a file browser and look at the _members.yml file. 
+## run the command below (with the right commit id) and then look at the file again.
+git checkout c1361 ## this takes you back in time to the state at this commit id. 
+
+## you are now in the legendary DETACHED HEAD state.
+## If you make any changes and commit them here, you'll need to
+## create a branch so that you can find them back later. 
+## Otherwise they'll eventually disappear forever.
+## git checkout --help has details about this. 
+
+## -- = anything after this is a file.
+## So basically, this command will take the file from c1361 and bring
+## it into your current location... reviving a file from the
+## dead! This takes ONLY the file members.yml, and no other changes. 
+git checkout c1361 -- members.yml
+
+git status ## will show the members.yml file as modified, 
+## since now you have a different version than the last commit.
+
+## If you want to go back to the state after the latest commit:
+git checkout -- members.yml 
+## since we didn't specify a commit, this assumes the latest commit, and give you that version
+
+## If you like the old version better:
+git commit -m "I'm keeping the old version"
+
+## push = upload your git repo to GitHub, basically syncing it.
+git push
+
+## remote add = set another GitHub to download from.  This remote is
```

Comment:
another repo (that happens to be on GitHub)

Reply:
Yes, I think this is important to clarify in order for students to understand that Git is a local program that could be used with any compatible online repository hub.

Reply:
May be I spend some time explaining this during the remote configuration part of the code along?